tests/scripts/rm_hpo_api_tests.sh

```diff
@@ -328,8 +333,164 @@ function other_post_experiment_tests() {
+		((TESTS++))
+		error_message ${failed}
+	fi
+}
+
+# Post different results for the same id & trial number
+function post_same_id_different_exp_result() {
+	# Post a valid experiment to RM-HPO /experiment_trials API.
+	exp="valid-experiment"
+	create_post_exp_json_array ${current_id}
+	post_experiment_json "${rm_hpo_post_experiment_json[$exp]}"
+	trial_num="${response}"
+
+	if [ "${http_code}" == "200" ]; then
+		failed=0
+
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 5
+
+		# Post a valid experiment result to RM-HPO /experiment_trials API.
+		exp_result="valid-experiment-result"
+		echo -n "Post the experiment result to HPO..."
+		create_post_exp_result_json_array ${current_id} ${trial_num}
+		post_experiment_trial "${rm_hpo_post_exp_result_json[$exp_result]}"
+
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 5
+
+		# Post a different valid experiment result for the same id and trial number to RM-HPO /experiment_trials API.
+		exp_result="valid-different-result"
+		echo -n "Post the differnt experiment result to HPO again for the same id and trial number..."
+		create_post_exp_result_json_array ${current_id} ${trial_num}
+		post_experiment_trial "${rm_hpo_post_exp_result_json[$exp_result]}"
+
+		actual_result="${http_code}"
+		expected_result_="^4[0-9][0-9]"
+		expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+
+		compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+	else
+		failed=1
+		expected_result_="200"
+		expected_behaviour="RESPONSE_CODE = 200 OK"
+		((TOTAL_TESTS++))
+		((TESTS++))
+		error_message ${failed}
+	fi
+}
+
+# Other RM-HPO post result tests
+function other_exp_result_post_tests() {
+	__test_name__=$1
+
+	for operation in "${other_exp_result_post_tests_[@]}"
+	do
+		TESTS_="${TEST_DIR}/${operation}"
+		mkdir -p ${TESTS_}
+		LOG_="${TEST_DIR}/${operation}.log"
+		echo "************************************* ${operation} Test ****************************************" | tee -a ${LOG_} ${LOG}
+
+		# Start the HPO servers
+		${SCRIPTS_DIR}/start_hpo_servers.sh -p ${TESTS_} | tee -a ${LOG_} ${LOG}
+
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 2
+
+		# Check if the servers have started
+		check_server_status
+
+		# Get the id from search space JSON
+		current_id=$(cat ${SEARCH_SPACE_JSON} | jq .[].id | tr -d '""')
+
+		operation=$(echo ${operation//-/_})
+		${operation}
+		echo ""
+
+		# Stop the HPO servers
+		${SCRIPTS_DIR}/start_hpo_servers.sh -t | tee -a ${LOG_} ${LOG}
+
+		# Sleep for few seconds to reduce the ambiguity
+		sleep 2
+
+		echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}
+	done	
+}
+
 # Tests for RM-HPO POST experiment
 function rm_hpo_post_experiment() {
-	invalid_post ${FUNCNAME}
+	run_post_tests ${FUNCNAME}
 	other_post_experiment_tests ${FUNCNAME}
 }
+
+# Tests for RM-HPO POST experiment
+function rm_hpo_post_exp_result() {
+	valid_exp="valid-experiment"
```

Comment:
Initialize these variables in the functions where they are used. Also add comments on why these are set to valid experiments or valid exp results